src/search_cuda.cpp

```diff
@@ -164,48 +166,78 @@ void GreedySearch_Cuda::SelectTop() {
   AppendNextTokensToSequences();
 }
 
-// TODO: Find a good way to do this on the GPU
-void SoftMax(std::span<float> scores, float temperature);
-void TopPSampling(int32_t* d_next_token, float* d_scores, int size, float threshold, float temperature) {
-  auto scores_buffer = CudaMallocHostArray<float>(size);
-  std::span<float> scores{scores_buffer.get(), static_cast<size_t>(size)};
-  cudaMemcpy(scores.data(), d_scores, size * sizeof(float), cudaMemcpyDeviceToHost);
-
-  SoftMax(scores, temperature);
-
-  // Sort an array of indices into the scores
-  std::vector<int32_t> indices(scores.size());
-  std::iota(indices.begin(), indices.end(), 0);
-  std::sort(indices.begin(), indices.end(), [scores = scores.data()](int32_t i, int32_t j) { return scores[i] > scores[j]; });
-
-  int32_t token = 0;
-  // Find the first token where the cumulative probability exceeds the threshold
-  for (int i = 0; i < scores.size(); i++) {
-    threshold -= scores[indices[i]];
-    if (threshold > 0)
-      continue;
-
-    token = indices[i];
-    break;
-  }
+// TODO: commented out in case of benchmarking
```

Comment:
> TODO: commented out in case of benchmarking [](http://example.com/codeflow?start=2&length=44)

do we need this function? if we need, we should uncomment it, or we need to remove those code

Reply:
if you just want to use it as a reference of performance, we don't need it after benchmark.